rocketpy/mathutils/function.py

```diff
@@ -2198,7 +2312,9 @@ def __rpow__(self, other):
             A Function object which gives the result of other(x)**self(x).
         """
         # Check if Function object source is array and other is float
-        if isinstance(other, (float, int, complex)):
+        if isinstance(
+            other, (float, int, complex, np.ndarray, np.integer, np.floating)
+        ):
```

Comment:
This comparison is repeated 6 times in this file.

Maybe we should define a constant at the top of this file containing a tuple of scalar classes, this way we prevent from modifying 6 code lines in the future if needed. Only a single code line would contin the data used in the logic.

Reply:
I agree, my suggestion would be creating a class variable, e.g. `ARITHMETIC_TYPES`.

In another opportunity we could also double check if there are no redundancies in these types and if there is no other way to support arithmetic with `np` types besides enumerating all of them.